app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -78,22 +78,27 @@ public ArrayList<Integer> getRemovedItems() {
         return removedItems;
     }
 
-    private MediaDetailPagerFragment() {}; // Constructor calls made to be explicit
-    @SuppressLint("ValidFragment")
-    public MediaDetailPagerFragment(Boolean editable, boolean isFeaturedImage) {
-        this.editable = editable;
-        this.isFeaturedImage = isFeaturedImage;
-        isFromFeaturedRootFragment = false;
-    }
 
-    @SuppressLint("ValidFragment")
-    public MediaDetailPagerFragment(Boolean editable, boolean isFeaturedImage, int position) {
-        this.editable = editable;
-        this.isFeaturedImage = isFeaturedImage;
-        isFromFeaturedRootFragment = true;
-        this.position = position;
+    /**
+     * This method will create a new instance of MediaDetailPagerFragment and the arguments will be
+     * saved to a bundle which will be later available in the {@link #onCreate(Bundle)}
+     * @param editable
+     * @param isFeaturedImage
+     * @return
+     */
+    public static MediaDetailPagerFragment newInstance(boolean editable, boolean isFeaturedImage) {
+        MediaDetailPagerFragment mediaDetailPagerFragment = new MediaDetailPagerFragment();
+        Bundle args = new Bundle();
+        args.putBoolean("is_editable", editable);
+        args.putBoolean("is_featured_image", isFeaturedImage);
+        mediaDetailPagerFragment.setArguments(args);
+        return mediaDetailPagerFragment;
     }
 
+    private MediaDetailPagerFragment() {}; // Constructor calls made to be explicit
```

Comment:
If I understand the issue description correctly, we need a public no-arg constructor. [Here's](https://stackoverflow.com/a/51831137/18659567) another explanation for it. I may have overlooked it but do we have any other no-arg constructor for this fragment that is public?

Reply:
@RitikaPahwa4444 Yes, you are right. The default constructor should be public. I will update the PR

Reply:
@RitikaPahwa4444  @nicolas-raoul Updated the PR with the necessary changes. Can you please review.